### PR TITLE
Updating markup for new Neue modal syntax.

### LINKED ISSUE #1078: modals - refine modal experience on iOS
Ideally we can make some tweaks to modals on iOS so that we can limit some of the user's ability to scroll outside of the modal.

Awkwardness can be seen by activating nav and/or going through site reg and scrolling down.

### LINKED ISSUE #1673: [IE Bug] Add background fallback for modal lightbox.
screen shot 2014-04-07 at 2 40 07 pm

rgba() not supported in IE8 so need to add a fallback. Either a transparent repeatable png, or a solid background color.

### PR DESCRIPTION
**Pull request by DFurnes**
Updating modal markup for DoSomething/neue#320. Adds analytics code that used to exist in Neue into the DS repo.

Fixes #1673. Fixes #1078.

### CODE REVIEW

File: lib/modules/dosomething/dosomething_helpers/modal-links.tpl.php [hunk 1]
```diff
@@ -1,21 +1,21 @@
 <ul class="modal-links">
   <?php if (isset($modals['faq'])): ?>
-    <li><a href="#modal-faq" class="js-modal-link">Check out our FAQs</a></li>
+    <li><a href="#" data-modal-href="#modal-faq">Check out our FAQs</a></li>
```
Comment (tjrosario) [inline lib/modules/dosomething/dosomething_helpers/modal-links.tpl.php:3]: What is the reason for using a data attribute for the 'href' ?

**Comment by DFurnes** [comment]
Tested the new modals with a bunch of random browser versions (including IE8-IE11) on BrowserStack and everything seems to work fine.

File: lib/modules/dosomething/dosomething_helpers/modal-links.tpl.php (hunk 1 above)
Comment (DFurnes) [inline lib/modules/dosomething/dosomething_helpers/modal-links.tpl.php:3]: The idea is that we generally want to have a canonical link (i.e. "Login" button goes to `/user/login` when JS is disabled), and that's what the `href` should point to.

In hindsight, I don't know if that's ever going to happen for a bunch of these links on the campaign page so maybe we should look into something more like the old syntax as an alternative way of linking to a modal.

**Comment by weerd** [comment]
Looks good :+1: :tractor:

**Pull request merged**